File: full-ver/scripts/anonymize_csv.py
#!/usr/bin/env python3
"""
@what CSV内の機密データをダミーデータに置き換える
@why 実際の料金や実績データを公開リポジトリに含めないため
"""

import csv
import random
import os

csv_path = os.path.join(os.path.dirname(__file__), '../demo-data-source/master_sample.csv')

print(f'📂 Reading CSV from: {csv_path}')

def random_price(min_val, max_val):
    return f'{random.randint(min_val, max_val):,}'

def random_clicks():
    return f'{random.randint(500, 10000):,}'

def random_achievements():
    num = random.randint(1, 100)
    return f'案件実績{num}件'

def dummy_remark():
    templates = [
        '①20代〜30代の美容関心層に影響力あり\n②過去の案件実績多数\n③エンゲージメント率が高い',
        '①ファッション・ライフスタイル層にリーチ\n②コスメ案件で高い成果\n③フォロワーとの信頼関係が強い',
        '①若年層への訴求力が高い\n②アクティブなフォロワー多数\n③ブランド親和性が高い',
        '①幅広い年齢層にリーチ可能\n②複数ジャンルで実績あり\n③継続案件での成果実績',
    ]
    return random.choice(templates)

def dummy_price_range():
    ranges = ['50-100万円', '100-200万円', '30-50万円', '20-30万円', '10-20万円']
    return random.choice(ranges)

# Read CSV with proper handling of multi-line fields
rows = []
with open(csv_path, 'r', encoding='utf-8') as f:
    reader = csv.reader(f)
    for row in reader:
        rows.append(row)

print(f'✅ Found {len(rows)} rows')

# Process rows
processed_rows = []
for i, row in enumerate(rows):
    # Skip header rows (first 2 rows)
    if i < 2:
        processed_rows.append(row)
        continue

    # Skip if not enough columns
    if len(row) < 26:
        print(f'⚠️  Row {i + 1} has only {len(row)} columns, skipping')
        processed_rows.append(row)
        continue

    # Replace sensitive data (0-indexed)
    # Column 11-13: 出稿費（税抜）
    row[10] = random_price(500000, 3000000)
    row[11] = random_price(300000, 2000000)
    row[12] = random_price(400000, 2500000)

    # Column 17: 想定クリック数
    row[16] = random_clicks()

    # Column 18: 案件実績
    row[17] = random_achievements()

    # Column 21: 備考
    row[20] = dummy_remark()

    # Column 26: 獲得できたオファー価格帯
    row[25] = dummy_price_range()

    processed_rows.append(row)

# Write back
with open(csv_path, 'w', encoding='utf-8', newline='') as f:
    writer = csv.writer(f)
    writer.writerows(processed_rows)

print('✅ Anonymization complete!')
print(f'📝 Processed {len(rows) - 2} data rows')
print(f'💾 Written to: {csv_path}')

File: full-ver/scripts/anonymize-csv.js
#!/usr/bin/env node
/**
 * @what CSV内の機密データをダミーデータに置き換える
 * @why 実際の料金や実績データを公開リポジトリに含めないため
 */

import fs from 'node:fs';
import path from 'node:path';
import { fileURLToPath } from 'node:url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const csvPath = path.resolve(__dirname, '../demo-data-source/master_sample.csv');

console.log('📂 Reading CSV from:', csvPath);
const content = fs.readFileSync(csvPath, 'utf-8');
const lines = content.split('\n');

console.log(`✅ Found ${lines.length} lines`);

// Random number generators
function randomPrice(min, max) {
  return Math.floor(Math.random() * (max - min + 1)) + min;
}

function randomClicks() {
  return Math.floor(Math.random() * 10000) + 500;
}

function randomAchievements() {
  const num = Math.floor(Math.random() * 100) + 1;
  return `${num}件`;
}

function dummyRemark() {
  const templates = [
    '①20代〜30代の美容関心層に影響力あり\n②過去の案件実績多数\n③エンゲージメント率が高い',
    '①ファッション・ライフスタイル層にリーチ\n②コスメ案件で高い成果\n③フォロワーとの信頼関係が強い',
    '①若年層への訴求力が高い\n②アクティブなフォロワー多数\n③ブランド親和性が高い',
    '①幅広い年齢層にリーチ可能\n②複数ジャンルで実績あり\n③継続案件での成果実績',
  ];
  return templates[Math.floor(Math.random() * templates.length)];
}

function dummyPriceRange() {
  const ranges = ['50-100万円', '100-200万円', '30-50万円', '20-30万円', '10-20万円'];
  return ranges[Math.floor(Math.random() * ranges.length)];
}

// Process lines
const processedLines = lines.map((line, index) => {
  // Skip header rows (first 2 lines)
  if (index < 2) {
    return line;
  }

  // Skip empty lines
  if (!line.trim()) {
    return line;
  }

  // Parse CSV line
  const cols = line.split(',').map(col => col.trim());

  if (cols.length < 26) {
    console.warn(`⚠️  Line ${index + 1} has only ${cols.length} columns, skipping`);
    return line;
  }

  // Replace sensitive data (0-indexed)
  // Column 11-13: 出稿費（税抜）
  cols[10] = randomPrice(500000, 3000000).toLocaleString(); // Story price
  cols[11] = randomPrice(300000, 2000000).toLocaleString(); // Feed price
  cols[12] = randomPrice(400000, 2500000).toLocaleString(); // Reel price

  // Column 17: 想定クリック数
  cols[16] = randomClicks().toLocaleString();

  // Column 18: 案件実績
  cols[17] = randomAchievements();

  // Column 21: 備考
  cols[20] = `"${dummyRemark()}"`;

  // Column 26: 獲得できたオファー価格帯
  cols[25] = dummyPriceRange();

  return cols.join(',');
});

// Write back
const outputContent = processedLines.join('\n');
fs.writeFileSync(csvPath, outputContent, 'utf-8');

console.log('✅ Anonymization complete!');
console.log(`📝 Processed ${lines.length - 2} data rows`);
console.log(`💾 Written to: ${csvPath}`);
